# Generated by Django 4.1.1 on 2022-09-19 15:02

from django.conf import settings
from django.db import migrations, models
import django.db.models.deletion


class Migration(migrations.Migration):

    dependencies = [
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
        ('blockso_app', '0005_rename_img_url_post_imgurl_and_more'),
    ]

    operations = [
        migrations.AlterField(
            model_name='follow',
            name='dest',
            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='follow_dest', to=settings.AUTH_USER_MODEL),
        ),
        migrations.AlterField(
            model_name='follow',
            name='src',
            field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='follow_src', to=settings.AUTH_USER_MODEL),
        ),
        migrations.AddConstraint(
            model_name='follow',
            constraint=models.UniqueConstraint(fields=('src', 'dest'), name='cannot follow twice constraint'),
        ),
        migrations.AddConstraint(
            model_name='follow',
            constraint=models.CheckConstraint(check=models.Q(('src__exact', models.F('dest')), _negated=True), name='cannot follow oneself constraint'),
        ),
    ]
